### PR TITLE
Simplify configuration of blockchains for example code plus add WS URI messaging manager interface

### PR DESCRIPTION
**Pull request by drinkcoffee**
* Simplify example code by passing around BlockchainConfig objects rather than individual pieces of configuration as many parameters. This also means code is more extensible, as more configuration can be put into blockchain configuration objects without needing to change code in dozens of files.

* Add ability to determine WS URI from MessagingManagerInterface. This is in preparation for Java SDK supporting event relay.

### CODE REVIEW

File: examples/sfc/erc20bridge/java/src/main/java/net/consensys/gpact/examples/sfc/erc20bridge/ERC20TokenBridgeExample.java [hunk 1]
```diff
@@ -64,34 +64,16 @@ public static void main(String[] args, boolean blockchainBIsMassConservation) th
...
       chainB =
           new MinterBurnerERC20Bridge(
-              "ChainB",
-              BigInteger.valueOf(CHAIN_B_TOKEN_SUPPLY),
-              erc20OwnerCreds,
-              bc2.bcId,
-              bc2.blockchainNodeRpcUri,
-              bc2.gasPriceStrategy,
-              bc2.period);
+              "ChainB", BigInteger.valueOf(CHAIN_B_TOKEN_SUPPLY), erc20OwnerCreds, bc2);
```
Comment (ermyas) [inline examples/sfc/erc20bridge/java/src/main/java/net/consensys/gpact/examples/sfc/erc20bridge/ERC20TokenBridgeExample.java:76]: is the `tokenSupply` required (or meaningful) for `MinterBurnerERC20Bridge` or only for `MassConservationERC20Bridge`? If not, should we remove this field from the abstract class and instead have it in the relevant subclass?

Comment (ermyas) [inline examples/sfc/erc20bridge/java/src/main/java/net/consensys/gpact/examples/sfc/erc20bridge/ERC20TokenBridgeExample.java:76]: is the `tokenSupply` required (or meaningful) for `MinterBurnerERC20Bridge` or only for `MassConservationERC20Bridge`? If not, should we remove this field from the abstract class and instead have it in the relevant subclass?

**Review by ermyas — APPROVED** [review]

File: examples/sfc/erc20bridge/java/src/main/java/net/consensys/gpact/examples/sfc/erc20bridge/ERC20TokenBridgeExample.java (hunk 1 above)
Comment (drinkcoffee) [reply examples/sfc/erc20bridge/java/src/main/java/net/consensys/gpact/examples/sfc/erc20bridge/ERC20TokenBridgeExample.java]: This field has been removed from the MinterBurner

Comment (drinkcoffee) [reply examples/sfc/erc20bridge/java/src/main/java/net/consensys/gpact/examples/sfc/erc20bridge/ERC20TokenBridgeExample.java]: This field has been removed from the MinterBurner

**Pull request merged**